core/jobs/src/main/java/com/cognifide/aet/job/common/modifiers/executejavascript/ExecuteJavaScriptModifier.java

```diff
@@ -39,26 +38,24 @@
   private static final String BASIC_AUTH_USERNAME = "basicAuthUsername";
   private static final String BASIC_AUTH_PASSWORD = "basicAuthPassword";
 
-  private final WebDriver webDriver;
   private final ExternalSnippetHttpClient httpClient;
+  private final JavaScriptJobExecutor jsExecutor;
 
   private String cmd;
   private String snippetUrl;
   private String basicAuth;
 
-  ExecuteJavaScriptModifier(WebDriver webDriver, ExternalSnippetHttpClient httpClient) {
-    this.webDriver = webDriver;
+  ExecuteJavaScriptModifier(ExternalSnippetHttpClient httpClient,
+      JavaScriptJobExecutor jsExecutor) {
     this.httpClient = httpClient;
+    this.jsExecutor = jsExecutor;
   }
 
   @Override
   public CollectorStepResult collect() throws ProcessingException {
-    final String jsSnippet = getJsSnippet();
-    if (LOG.isDebugEnabled()) {
-      LOG.debug("Executing JavaScript command: {} on page: {}", jsSnippet,
-          webDriver.getCurrentUrl());
-    }
-    return javaScriptElement(webDriver, jsSnippet);
+    String jsSnippet = getJsSnippet();
+    jsExecutor.execute(jsSnippet);
```

Comment:
Now this method can throw `ProcessingException`, so we need to add a catch block as before and return `CollectorStepResult.newProcessingErrorResult(message);` if exception is thrown

Reply:
OK, I just thought it looks cleaner this way and does not change much. Fixed!